benchmarks/src/main/java/io/grpc/benchmarks/SocketAddressValidator.java

```diff
@@ -31,12 +31,14 @@
 
 package io.grpc.benchmarks;
 
+import com.google.errorprone.annotations.Immutable;
 import java.net.InetSocketAddress;
 import java.net.SocketAddress;
 
 /**
  * Verifies whether or not the given {@link SocketAddress} is valid.
  */
+@Immutable
 public interface SocketAddressValidator {
```

Comment:
Pro enum: 
* Effective Java 19 says so, and it's usually right.
* It's probably slightly faster, cleaner, more readable.

Pro leave-this-as-is:
* It's in benchmarks, and HK?

I am ambivalent.  Since neither is objectively better than the other, I can go either way.  Right of way goes to the author so Eric will need to decide what's best long term for gRPC.  If pressed to a decision, I count myself in the clean-things-up-as-you-go camp and would pick an enum as well.